minio/api.py

```diff
@@ -1355,7 +1396,12 @@ def presigned_url(self, method,
                                        ' {0} secs'.format(_MAX_EXPIRY_TIME))
 
         region = self._get_bucket_region(bucket_name)
-        url = get_target_url(self._endpoint_url,
+        endpoint_url = self._endpoint_url
+        if self._enable_s3_accelerate:
+            if self.is_bucket_accelerate_configured(bucket_name):
```

Comment:
This type of code is expensive when transfer acceleration is not enabled.  You don't need to implement the verification for parser acceleration and let the server fail.

Reply:
So I am specifically checking this because acceleration is enabled at global level and not per bucket. 

If a bucket is not configured to use acceleration it uses the normal url. Eg. some internal bucket ("bucket A") where acceleration is disabled but for user uploads ("bucket B") acceleration is configured. Using the Minio i want to access both the buckets, but use acceleration when possible, if it is enabled in Minio.

Reply:
Let it fail on the server side.  I don't think we should have such type of checks.

Reply:
Hey, I have removed the acceleration check